### ISSUE TITLE
[FEAT] Honor port in autofill

### ISSUE THREAD

**Issue by githubomane** [PhilippC/keepass2android#2721]
Scenario:
I have a server which has different websites avaliable at differing port, e. g. Portainer at myserver:8000 and a rss-Reader at myserver:8100. The credentials for my portainer and my rss-reader are stored in keepass with the url myserver:port. 

Current behaviour:
At the login page of Portainer at myserver:8000, the credentials of my rss-reader are offered for auto fill. When choosing "Fill with keepass2android", both the credentials for portainer and the rss-reader are offered. 

Expected behaviour:
Keepass2android should auto fill credentials only when the port of the stored url matches the port of the website. 

Used apps:
Firefox Android 133.0 // Kiwi Browser 124.0.6327.4
Keepass2android 1.11-r0

**Comment by PhilippC** [comment]
I agree that this is not good. As a workaround: If you can add the exact URLs of each login page, this should help.

The logic is like this:
 * search for entry/entries with exact URL match
 * if nothing found: search for entry/entries with the hostname (including subdomain) matching. (Here the current implementation throws the port away which is not good)
 * if nothing found:search for entry/entries with the hostname (without subdomain) matching (("www.google.com" in entry is ok for "accounts.google.com" in search (but not the other way around))
* if no results returned up to now, try to search through other fields for the URL
* if no results returned up to now, try to search through other fields for the hostname
